examples/system_tray.rs

```diff
@@ -173,6 +148,21 @@ fn main() {
   });
 }
 
+#[cfg(any(target_os = "windows", target_os = "linux", target_os = "macos"))]
+#[cfg(any(feature = "tray", all(target_os = "linux", feature = "ayatana")))]
+fn load_icon_from_slice(slice: &[u8]) -> tao::system_tray::Icon {
+  let (icon_rgba, icon_width, icon_height) = {
+    let image = image::load_from_memory(slice)
+      .expect("Failed to parse icon slice")
+      .into_rgba8();
+    let (width, height) = image.dimensions();
+    let rgba = image.into_raw();
+    (rgba, width, height)
+  };
+  tao::system_tray::Icon::from_rgba(icon_rgba, icon_width, icon_height)
+    .expect("Failed to open icon")
+}
```

Comment:
Looks like icon on mac need to use origin `slice` as NSData instead of rgba's
Maybe we should move rgba creation to linux and windows inside.

Reply:
This would defeat the purpose of this PR, I will convert this PR to draft for now.

Reply:
pushed a commit to use the original slice.